File: JULY-2025/16-07-2025/p3201.kt
class Solution {
    fun maximumLength(nums: IntArray): Int {
        var evenCount = 0;
        var oddCount = 0;
        var altCount = 0;
        var lastAlt = -1;

        for (i in 0 until nums.size){
            if (nums[i] % 2 == 0){
                evenCount++;
            }
            else {
                oddCount++;
            }
            if (lastAlt == -1 || nums[lastAlt] % 2 != nums[i] % 2) {
                altCount++;
                lastAlt = i;
            }
        }

        return Math.max(Math.max(evenCount, altCount), oddCount);
    }
}

File: JULY-2025/16-07-2025/p3201.swift
class Solution {
    func maximumLength(_ nums: [Int]) -> Int {
        var evenLen = 0, oddLen = 0, altLen = 0, altParity = -1
        for num in nums {
            if num % 2 == 0 {
                evenLen += 1
                if altParity != 0 { altLen += 1 }
            } else {
                oddLen += 1
                if altParity != 1 { altLen += 1 }
            }
            altParity = num % 2
        }
        return max(evenLen, oddLen, altLen)
    }
}